### PR TITLE
Remove deprecated packaging and cython

### PR DESCRIPTION
**Pull request by twizmwazin**
**Your checklist for this pull request**
- [x] I've documented or updated the documentation of every API function and struct this PR changes.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)

**Detailed description**

Cherry-picks #2396 onto v5

**Test plan**

...

**Closing issues**

closes https://github.com/capstone-engine/capstone/issues/2005
closes https://github.com/capstone-engine/capstone/issues/2341
closes https://github.com/capstone-engine/capstone/issues/2223

### CODE REVIEW

**Comment by twizmwazin** [comment]
Alright I think this is ready.

**Comment by twizmwazin** [comment]
So this branch still uses the old wheel building method. I'm gonna copy over the changes for cibuildwheel from `next`.

File: bindings/python/tests/test_aarch64.py
Comment (Rot127) [inline bindings/python/tests/test_aarch64.py:1]: Remove this one, please. It is effectively not used. In `v5` the name is still `test_arm64`

File: bindings/python/tests/test_alpha.py
Comment (Rot127) [inline bindings/python/tests/test_alpha.py:1]: Also, this one. And for HPPA below.

File: bindings/python/tests/test_iter.py
Comment (Rot127) [inline bindings/python/tests/test_iter.py:1]: `test_iter` is nowhere used as well, right?

**Review by Rot127 — APPROVED** [review]
Please just remove the unnecessary files. Then this is good to go in my opinion.

**Review by Rot127 — APPROVED** [review]
@kabeor

**Comment by XVilka** [comment]
Please add:
closes https://github.com/capstone-engine/capstone/issues/2005
closes https://github.com/capstone-engine/capstone/issues/2341
closes https://github.com/capstone-engine/capstone/issues/2223

**Review by kabeor — APPROVED** [review]
LGTM, Thanks!

**Comment by Rot127** [comment]
@twizmwazin 
Unfortunately this broke the CI:

The v5 branch is red
There seems to be a left over CI job?

https://github.com/capstone-engine/capstone/actions/runs/10071537984/job/27841854148
https://github.com/capstone-engine/capstone/actions/runs/10071537984/job/27841854447

Do you think you will find time to look at it?
Otherwise I try to fix it. But will take longer.

**Pull request merged**